File: task-filter/java/src/test/java/study/gradle/taskfilterjava/HelloServiceIntegrationTest.java
package study.gradle.taskfilterjava;

import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

@SpringBootTest
public class HelloServiceIntegrationTest {

  @Autowired
  private HelloService sut;

  @Test
  public void greetingIntegration() {
    var result = sut.greeting();

    Assertions.assertThat(result).isEqualTo("hello");
  }
}

File: task-filter/kotlin/src/test/kotlin/study/gradle/taskfilterkotlin/HelloServiceIntegrationTest.kt
package study.gradle.taskfilterkotlin

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

class HelloServiceIntegrationTest : IntegrationTest() {

    @Autowired
    private lateinit var sut: HelloService

    @Test
    fun greetingIntegration() {
        val result = sut.greeting()

        Assertions.assertThat(result).isEqualTo("hello")
    }
}